src/control/xml/XmlNode.cpp

```diff
@@ -72,7 +72,9 @@ void XmlNode::setAttrib(const char* attrib, size_t  value)
  */
 void XmlNode::setAttrib(const char* attrib, double* value, int count)
 {
-	putAttrib(new DoubleArrayAttribute(attrib, value, count));
+	std::vector<double> points{value, value + count};
+	g_free(value);
```

Comment:
as a Todo for the future; do this directly on the calling side.

Reply:
We should create a new issue for modernizing the serialization code (but it may become obsolete when we start working on the new file format).

Reply:
Yep working on this has absolutely no use. The new file format will provide its own code to export to xopp and big parts of our codebase are not really reusable for this. It will be more like picking a few lines here and there to get the new library running but no copying of files/algorithms.